store/store.go

```diff
@@ -168,23 +173,12 @@ func (s *Store[H]) Height() uint64 {
 }
 
 func (s *Store[H]) Head(ctx context.Context, _ ...header.HeadOption[H]) (H, error) {
-	head, err := s.GetByHeight(ctx, s.heightSub.Height())
-	if err == nil {
-		return head, nil
+	if head := s.contiguousHead.Load(); head != nil {
+		return *head, nil
 	}
 
 	var zero H
-	head, err = s.readHead(ctx)
-	switch {
-	default:
-		return zero, err
-	case errors.Is(err, datastore.ErrNotFound), errors.Is(err, header.ErrNotFound):
-		return zero, header.ErrNoHead
-	case err == nil:
-		s.heightSub.SetHeight(head.Height())
-		log.Infow("loaded head", "height", head.Height(), "hash", head.Hash())
-		return head, nil
-	}
+	return zero, header.ErrNoHead
```

Comment:
Should we return a dedicated error here? `ErrNoHead` might be too broad, from another side getting `ErrNoHead` from `Head` means only one case.

Reply:
Resolved in threads below.